disttask/framework/dispatcher/dispatcher.go

```diff
@@ -277,7 +277,14 @@ func (d *dispatcher) detectTask(gTask *proto.Task) {
 					zap.Int64("taskID", gTask.ID), zap.String("state", gTask.State))
 				break
 			}
-
+			// process finish flow if task finish a step.
+			if stepIsFinished && len(errStr) == 0 && gTask.State == proto.TaskStateRunning {
+				logutil.BgLogger().Info("detect task, this task finished a step",
+					zap.Int64("taskID", gTask.ID), zap.String("state", gTask.State), zap.Int64("step", gTask.Step))
+				if err := d.processFinishFlow(gTask); err != nil {
```

Comment:
I think we can put `processFinishFlow` directly, and only import step2 or other special step needs to do this.